From 4da373e3ef8e9bf5488000658b80ce0f3cc69ade Mon Sep 17 00:00:00 2001
From: Alisher Alikhodjaev <alisher@google.com>
Date: Wed, 1 Mar 2023 12:50:11 -0800
Subject: [PATCH] OOBR in AnalyzeMfcResp in NxpMfcReader.cc

Bug: 252763983
Test: build ok
Merged-In: I91fa035ca6245e6039eeedb447d7e3306b7aebc5
Change-Id: I20b718a74afd2f4de23ee59716601f82190ea3f0
(cherry picked from commit on googleplex-android-review.googlesource.com host: 67e1db04f568cf50c90758272ceca93426aba932)
Merged-In: I20b718a74afd2f4de23ee59716601f82190ea3f0
---
 halimpl/mifare/NxpMfcReader.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/halimpl/mifare/NxpMfcReader.cc b/halimpl/mifare/NxpMfcReader.cc
index 38b6bb5..5e4873e 100644
--- a/halimpl/mifare/NxpMfcReader.cc
+++ b/halimpl/mifare/NxpMfcReader.cc
@@ -345,6 +345,10 @@ NFCSTATUS NxpMfcReader::AnalyzeMfcResp(uint8_t *pBuff, uint16_t *pBufflen) {
     } break;
 
     case eMfcAuthRsp: {
+      if (*pBufflen < 2) {
+        status = NFCSTATUS_FAILED;
+        break;
+      }
       /* check the status byte */
       if (NFCSTATUS_SUCCESS == pBuff[1]) {
         status = NFCSTATUS_SUCCESS;
-- 
2.40.0.rc1.284.g88254d51c5-goog

